### PR TITLE
[DOCS-8376 / 8426] OP Add components and set up pipelines docs

### PR DESCRIPTION
**Pull request by maycmlee**
### What does this PR do? What is the motivation?

- Adds these new doc:
  - Set Up Pipelines
  - Sources and child pages
  - Processors
  - Destinations
- Restructures the nav so that there's a 4th level for Set Up Pipelines 

Will open a separate PR later to move all use case source docs under Set Up Pipelines folder.

### Merge instructions

- [ ] Please merge after reviewing

### CODE REVIEW

**Review by jhgilbert — APPROVED** [review]
Approving with suggestions but happy to take another look anytime!

File: content/en/observability_pipelines/sources/sumo_logic.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+---
+title: Sumo Logic Hosted Collector
+disable_toc: false
+aliases:
+- /path-to-old-doc/
```
[the remark this answers is not in the document]
Comment (maycmlee) [reply content/en/observability_pipelines/sources/sumo_logic.md]: Deleted

File: content/en/observability_pipelines/sources/syslog.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+---
+title: Syslog Source
+disable_toc: false
+aliases:
+- /path-to-old-doc/
```
[the remark this answers is not in the document]
Comment (maycmlee) [reply content/en/observability_pipelines/sources/syslog.md]: Deleted

File: content/en/observability_pipelines/sources/_index.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+disable_toc: false
+further_reading:
+- link: "logs/processing/pipelines"
+  tag: "Documentation"
+  text: "Log processing pipelines"
+---
+
+## Overview
+
+Use Observability Pipelines's sources to receive logs from your different log sources.
```
[the remark this answers is not in the document]
Comment (maycmlee) [reply content/en/observability_pipelines/sources/_index.md:12]: I think I'll update it to just `'` to be consistent with everything else.

**Review by fitzage — APPROVED** [review]

**Pull request merged**